lib/command/apply_template.rb

```diff
@@ -87,24 +65,83 @@ def call # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, M
 
     private
 
-    def templates
-      @templates ||= config.args.to_h do |template|
-        [template, "#{config.app_cpln_dir}/templates/#{template}.yml"]
+    def template_filename(name)
+      "#{config.app_cpln_dir}/templates/#{name}.yml"
+    end
+
+    def template_kind(template)
+      case template["kind"]
+      when "gvc" then "app"
+      else template["kind"]
       end
     end
 
+    def template_duplicate?(template1, template2)
+      template1["kind"] == template2["kind"] && template1["name"] == template2["name"]
+    end
+
     def ensure_templates!
-      missing_templates = templates.reject { |_template, filename| File.exist?(filename) }.to_h
+      missing_templates = config.args.reject { |name| File.exist?(template_filename(name)) }
       return if missing_templates.empty?
 
-      missing_templates_str = missing_templates.map do |template, filename|
-        "  - #{template} (#{filename})"
+      missing_templates_str = missing_templates.map do |name|
+        "  - #{name} (#{template_filename(name)})"
       end.join("\n")
       progress.puts("#{Shell.color('Missing templates:', :red)}\n#{missing_templates_str}\n\n")
 
       raise "Can't find templates above, please create them."
     end
 
+    def parse_templates # rubocop:disable Metrics/MethodLength
+      config.args.each_with_object([]) do |name, templates|
+        data = File.read(template_filename(name))
+        data = replace_variables(data)
+        templates_data = data.split(/^---\s*$/)
+        templates_data.each do |template_data|
+          template = YAML.safe_load(template_data)
```

Comment:
Before, we made assumptions about templates, like assuming that only a file named `gvc.yml` could be for an app, or that any other file would be for a workload.

Now, by parsing each item in each file separately, we can check its `kind` and be sure of what is an app and what is a workload.

And now we can use `YAML.safe_load`.